### PR TITLE
chore(EMI-2025): Add tracking for Limited-Time Offer signals

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [EMI-2025] 

### Description
This PR adds tracking for Limited-Time Offer signal tracking
⚠️ It also updates some of the deprecated tracking functions for artwork grid item

Please let me know if you have any concerns or questions about my code, naming, or ways of adding fields :) 

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [X] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [X] I have added **tests**, or my changes don't require any.
- [X] I added an **[app state migration]**, or my changes do not require one.
- [X] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [X] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Update tracking for artwork taps + add tracking for Limited-Time Offer signals - mrsltun

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[EMI-2025]: https://artsyproduct.atlassian.net/browse/EMI-2025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Update tracking for artwork taps + add tracking for Limited-Time Offer signals - mrsltun)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 95b2f88e3d39aa7ff8a4e1f7b490e4b0453b7a61
</p>

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx [hunk 1]
```diff
@@ -208,18 +208,30 @@ export const Artwork: React.FC<ArtworkProps> = ({
...
+        type: "thumbnail",
+      }
+
+      if (
+        AREnablePartnerOfferSignals &&
+        collectorSignals?.partnerOffer?.isAvailable &&
+        !artwork.sale?.isAuction
+      ) {
+        genericTapEvent.signal_label = "Limited-Time Offer"
+      }
```
Comment (erikdstock) [inline src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:233]: As of today partner offer + auction artwork are mutually exclusive so this check should not be necessary.

Comment (erikdstock) [reply src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:233]:
```suggestion
      if (
        AREnablePartnerOfferSignals &&
        !!collectorSignals?.partnerOffer
      ) {
        genericTapEvent.signal_label = "Limited-Time Offer"
      }
```
Technically this should be all we need - `isAvailable` refers to the artwork being available for purchase, and the partner offer is only returned if the artwork is purchaseable and the partner offer is still active.

**Review by erikdstock — COMMENTED** [review]
LGTM, one minor comment where I think this could be simpler.

Non-blocking, it seems like a smell if we need to determine the signal label both when sending the tap event and for rendering the element itself. Maybe we can eventually have a helper that takes an `artwork.collectorSignals` and outputs all the values for both rendering and analytics?

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx [hunk 2]
```diff
@@ -208,18 +208,30 @@ export const Artwork: React.FC<ArtworkProps> = ({
...
-        sort: String(filterParams?.sort),
         query: contextScreenQuery,
-      })
+        sort: filterParams?.sort,
+        type: "thumbnail",
+      }
+
+      if (
+        AREnablePartnerOfferSignals &&
+        collectorSignals?.partnerOffer?.isAvailable &&
```
[the remark this answers is not in the document]
Comment (MrSltun) [reply src/app/Components/ArtworkGrids/ArtworkGridItem.tsx]: I like your approach, but I'll update it in a follow-up PR :)

**Pull request merged**